VoiceProcessor.swift

```diff
@@ -43,12 +51,25 @@ public class VoiceProcessor {
         bytesPerFrame: UInt32 =  2,
         channelsPerFrame: UInt32 = 1,
         bitsPerChannel: UInt32 = 16,
-        reserved: UInt32 = 0
+        reserved: UInt32 = 0,
+        sessionErrorCallback: ((Error) -> Void)? = nil
     ) throws {
         if started {
             return
         }
         
+        do {
+            try self.session.setCategory(AVAudioSession.Category.playAndRecord, options: [.mixWithOthers, .defaultToSpeaker, .allowBluetooth])
+            try self.session.setActive(true, options: .notifyOthersOnDeactivation)
+        } catch {
+            if self.sessionErrorCallback != nil {
+                sessionErrorCallback!(error)
```

Comment:
I think in this case, we should just have the start function throw rather than have a user-provided callback. The callback pattern only really makes sense when we need to return an error from a separate thread.